src/otaclient/grpc/api_v2/servicer.py

```diff
@@ -225,6 +242,22 @@ async def rollback(
         logger.info(f"receive rollback request: {request}")
         response = api_types.RollbackResponse()
 
+        # NOTE(20241220): due to the fact that OTA Service API doesn't have field
+        #                 in UpdateResponseEcu msg, the only way to pass the failure_msg
+        #                 to upper is by status API.
+        if not otaclient_cfg.ECU_INFO_LOADED_SUCCESSFULLY:
+            logger.error(
+                "ecu_info.yaml is not loaded properly, reject any rollback request"
+            )
+            for _rollback_req in request.iter_ecu():
+                response.add_ecu(
+                    api_types.RollbackResponseEcu(
+                        ecu_id=_rollback_req.ecu_id,
+                        result=api_types.FailureType.UNRECOVERABLE,
+                    )
+                )
+            return response
+
```

Comment:
nit: Can we aggregate as a function between `update` and `rollback` ?

Reply:
Emmmm I will persist on splitting them into two methods though, for the single-responsibility principle.